src/witan/models/run_models.clj

```diff
@@ -0,0 +1,118 @@
+       ld/create-dataset-after-coercion
+       (hash-map keyname)))
+
+(defn get-datasets
+  "Input should be a map with keys for each dataset and filepaths to csv
+   files as the values. Output is a map of core.matrix datasets."
+  [file-map gss-code]
+  (->> file-map
+       (mapv (fn [[k path]] (get-dataset k path gss-code)))
+       (reduce merge)))
+
+;; Functions from witan.models.econscratch to write datasets
+;; to CSV and specify the order of the columns:
+
+(defn order-result-map
+  "Takes in a map of data and a map with colnames order.
+  Returns the map of ordered data."
+  [result-map order]
+  (into (sorted-map-by (fn [k1 k2]
+                         (compare [(get order k1) k1]
+                                  [(get order k2) k2])))
+        result-map))
+
+(defn get-sorted-colnames
+  "Takes a filepath to a csv file and outputs a map with
+  the rank for each column name based on the columns order in the file."
+  ([filename]
+   (get-sorted-colnames filename nil))
+  ([filename eol]
+   (let [normalised-data (-> (slurp filename)
+                             (str/replace "\r\n" "\n")
+                             (str/replace "\r" "\n"))
+         parsed-csv (csv/parse-csv normalised-data :end-of-line nil)
+         parsed-data (rest parsed-csv)
+         headers (mapv keyword (first parsed-csv))]
+     (zipmap headers
+             (vec (range (count headers)))))))
+
+(defn convert-row-maps-to-vector-of-vectors
+  "Takes a sequence of maps and returns a vector of vectors where the first vector
+   contains column names and the following vetors contain the value for each
+   row. This is the format needed to save as csv using the write-to-csv function"
+  [rows-as-maps]
+  (let [colnames (mapv name (keys (first rows-as-maps)))
+        rows (mapv #(vec (vals %)) rows-as-maps)]
+    (into [colnames] rows)))
+
+(defn write-data-to-csv
+  "Takes a dataset or coll of maps, a filepath for the output csv and a filepath for a csv OR
+  a vector (example [:column-1 :column-2 :column-3]) OR a map (example
+  {:column-1 0 :column-2 1 :column 3: 2} to use for the columns order.
+  Writes to csv file with the filepath provided, for example 'folder1/folder2/filename.csv'
+  or 'filename.csv'"
+  [data filepath ordered-colnames]
+  (let [rows-as-maps (cond (i/dataset? data) (ds/row-maps data)
+                           :else data)
+        sorted-colnames (cond (string? ordered-colnames)
+                              (get-sorted-colnames ordered-colnames)
+                              (map? ordered-colnames)
+                              ordered-colnames
+                              (vector? ordered-colnames)
+                              (zipmap ordered-colnames
+                                      (vec (range (count ordered-colnames)))))
+        ordered-data (map #(order-result-map % sorted-colnames) rows-as-maps)]
+    (with-open [out-file (jio/writer filepath)]
+      (data-csv/write-csv out-file
```

Comment:
I know clojure-csv.core and clojure.data.csv are each good in there own ways, but just wanted to double check we should be using both rather than just one? I know both have a `write-csv` fn